src/components/BasketballCourt/KeyArea.tsx

```diff
@@ -2,28 +2,32 @@ import { Rect } from "react-konva";
 import { useStoreSelector } from "@/store/hooks";
 import { courtWhiteLine } from "../../store/reducer/courtSizeSlice";
 import { ICourtStartPoint } from "@/interfaces/courtStartPoint";
+import { useState } from "react";
 
-interface KeyAreaTextProps {
+interface KeyAreaProps {
   startPoint: ICourtStartPoint;
 }
 
-const KeyArea: React.FC<KeyAreaTextProps> = ({ startPoint }) => {
+const KeyArea: React.FC<KeyAreaProps> = ({ startPoint }) => {
   const { threePointLineToCourtEdgeLenth, threePointLineRadius, keyAreaWidth, keyAreaHeight } =
     useStoreSelector((state) => state.courtSize);
 
-  const color = useStoreSelector(
-    (state) => state.tile.find((tile) => tile.location.includes("keyArea"))?.color
-  );
+  const selectedColor = useStoreSelector((state) => state.courtColor.color);
+  const [keyAreaColor, setKeyAreaColor] = useState("#2C4E8A");
 
+  const handleColorChange = () => {
+    setKeyAreaColor(selectedColor);
+  };
   return (
     <Rect
       width={keyAreaWidth}
       height={keyAreaHeight}
-      fill={color}
+      fill={keyAreaColor}
       stroke="white"
       strokeWidth={courtWhiteLine}
       x={startPoint.X}
       y={startPoint.Y + threePointLineToCourtEdgeLenth + threePointLineRadius - keyAreaHeight / 2}
+      onClick={() => handleColorChange()}

```

Comment:
if you won't pass any  props, you can simples use "onClick={handleColorChange}" 
you don't need need to change it,  just  a remider

Reply:
Thanks for the reminder!